subprojects/gradle/room-config/src/main/kotlin/com/avito/android/plugin/incremental/RoomIncrementalKaptSupportChecker.kt

```diff
@@ -0,0 +1,53 @@
+package com.avito.android.plugin.incremental
+
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask
+import java.net.URLClassLoader
+import javax.annotation.processing.AbstractProcessor
+
+internal class RoomIncrementalKaptSupportChecker(
+    private val project: Project
+) {
+
+    fun isSupported(): Boolean {
+        val processor = createRoomProcessor()
+        try {
+            return checkOfficialWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in official way: %e")
+        }
+
+        try {
+            return checkReflectionWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in reflection way: %e")
+        }
+
+        return false
+    }
+
+    private fun createRoomProcessor(): AbstractProcessor =
+        getKaptClassLoader().loadClass("androidx.room.RoomProcessor").newInstance() as AbstractProcessor
+
+
+    private fun getKaptClassLoader(): ClassLoader {
+        val task = project.tasks.findByName("kaptGenerateStubsReleaseKotlin") as KaptGenerateStubsTask
```

Comment:
What if a module doesn't have a release build variant?

Reply:
Not actual. Move this check from this PR to another task.